lib/withOnyx.d.ts

```diff
@@ -0,0 +1,23 @@
+import {IsEqual} from 'type-fest';
+import {Key, Selector, Value} from './types';
+
+type Mapping<TComponentProps, TOnyxProps, TMappingKey extends keyof TOnyxProps, TOnyxKey extends Key> = IsEqual<Value[TOnyxKey] | null, TOnyxProps[TMappingKey]> extends true
```

Comment:
I don't think we need to create the type parameter `TMappingKey`.

instead of `TOnyxProps[TMappingKey]`,
we can do `TOnyxProps[keyof TOnyxProps]`

Reply:
@hayata-suenaga This `Mapping` type is for each individual object of `{key, selector...}`, that's why I need `TMappingKey` to know which prop that individual object is related.

Reply:
ah I got it 👍